### ISSUE TITLE
While training with navtrain dataset, encountered an error "libcudnn_cnn_infer.so.8. Error: libcuda.so: cannot open shared object file: No such file or directory Aborted (core dumped)"

### ISSUE THREAD

**Issue by bskkimm** [hustvl/DiffusionDrive#36]
While trying to train with navtrain dataset, I'm getting the libcudnn_cnn_infer.so.8. Error: libcuda.so: cannot open shared object file: No such file or directory
Aborted (core dumped) error. 

anyone encountered this error?

Thank you!

`python $NAVSIM_DEVKIT_ROOT/navsim/planning/script/run_training.py \
        agent=diffusiondrive_agent \
        experiment_name=training_diffusiondrive_agent  \
        train_test_split=navtrain  \
        split=trainval   \
        trainer.params.max_epochs=100 \
        cache_path="${NAVSIM_EXP_ROOT}/training_cache/" \
        use_cache_without_dataset=True  \
        force_cache_computation=False `

--> 

`[2025-02-21 14:08:39,130][__main__][INFO] - Starting Training
You are using a CUDA device ('NVIDIA GeForce RTX 4070 Laptop GPU') that has Tensor Cores. To properly utilize them, you should set `torch.set_float32_matmul_precision('medium' | 'high')` which will trade-off precision for performance. For more details, read https://pytorch.org/docs/stable/generated/torch.set_float32_matmul_precision.html#torch.set_float32_matmul_precision
[rank: 0] Seed set to 0
Initializing distributed: GLOBAL_RANK: 0, MEMBER: 1/1
[2025-02-21 14:08:39,491][torch.distributed.distributed_c10d][INFO] - Added key: store_based_barrier_key:1 to store for rank: 0
[2025-02-21 14:08:39,491][torch.distributed.distributed_c10d][INFO] - Rank 0: Completed store-based barrier for key:store_based_barrier_key:1 with 1 nodes.
----------------------------------------------------------------------------------------------------
distributed_backend=nccl
All distributed processes registered. Starting with 1 processes
----------------------------------------------------------------------------------------------------

Missing logger folder: /home/bskim/Diffusion_Project/navsim_workspace/exp/training_diffusiondrive_agent/2025.02.21.14.08.31/lightning_logs
LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]

  | Name  | Type            | Params
------------------------------------------
0 | agent | TransfuserAgent | 60.7 M
------------------------------------------
60.7 M    Trainable params
320       Non-trainable params
60.7 M    Total params
242.863   Total estimated model params size (MB)
Epoch 0:   0%|                                                                  | 0/1330 [00:00<?, ?it/s]Could not load library libcudnn_cnn_infer.so.8. Error: libcuda.so: cannot open shared object file: No such file or directory
Aborted (core dumped)`

**Comment by LegendBC** [comment]
It seems that you have not install cudnn. https://www.digitalocean.com/community/tutorials/install-cuda-cudnn-for-gpu

**Comment by bskkimm** [comment]
> It seems that you have not install cudnn. https://www.digitalocean.com/community/tutorials/install-cuda-cudnn-for-gpu

Thank you for your rapid reply! I have another question. Should I install cuda toolkit manually or were some cuda-enabled libraries installed already when creating navsim and diffusion enviornment. If I have to install manually, should I specifically install cuda 11.7? Thank you for your help!

**Comment by bskkimm** [comment]
> It seems that you have not install cudnn. https://www.digitalocean.com/community/tutorials/install-cuda-cudnn-for-gpu

Sorry for consecutive questions. cudnn seems installed already when building navsim env.,,,

`Python 3.9.21 | packaged by conda-forge | (main, Dec  5 2024, 13:51:40) 
[GCC 13.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import torch
>>> import torch.backends.cudnn as cudnn
>>> cuda_available = torch.cuda.is_available()
>>> cudnn_available = cudnn.is_available()
>>> print(f"CUDA available: {cuda_available}")
CUDA available: True
>>> print(f"cuDNN available: {cudnn_available}")
**_cuDNN available: True_**
>>> cudnn_version = cudnn.version()
>>> print(f"cuDNN version: {cudnn_version}")
cuDNN version: 8500
>>> `

**Comment by LegendBC** [comment]
you may refer to this, https://discuss.pytorch.org/t/libcudnn-cnn-infer-so-8-library-can-not-found/164661/2

**Comment by bskkimm** [comment]
> you may refer to this, https://discuss.pytorch.org/t/libcudnn-cnn-infer-so-8-library-can-not-found/164661/2

Thank you for your kind help. The error has been solved by adding the below to the bashrc file
`export LD_LIBRARY_PATH=/usr/lib/wsl/lib:$LD_LIBRARY_PATH`
